disperser/batcher/orchestrator.go

```diff
@@ -0,0 +1,216 @@
+	encodingStreamer, err := NewEncodingStreamer(streamerConfig, queue, chainState, encoderClient, assignmentCoordinator, batchTrigger, encodingWorkerPool, metrics.EncodingStreamerMetrics, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	miniBatcher, err := NewMinibatcher(config.MinibatcherConfig, queue, minibatchStore, dispatcher, encodingStreamer, encodingWorkerPool, logger, metrics)
+	if err != nil {
+		return nil, err
+	}
+
+	batchConfirmerConfig := BatchConfirmerConfig{
+		PullInterval:                 config.PullInterval,
+		DispersalTimeout:             timeoutConfig.DispersalTimeout,
+		DispersalStatusCheckInterval: config.MinibatcherConfig.DispersalStatusCheckInterval,
+		AttestationTimeout:           timeoutConfig.AttestationTimeout,
+		SRSOrder:                     config.SRSOrder,
+		NumConnections:               config.MinibatcherConfig.MaxNumConnections,
+		MaxNumRetriesPerBlob:         config.MinibatcherConfig.MaxNumRetriesPerBlob,
+	}
+	batchConfirmer, err := NewBatchConfirmer(batchConfirmerConfig, queue, minibatchStore, dispatcher, chainState, assignmentCoordinator, encodingStreamer, aggregator, ethClient, transactor, txnManager, miniBatcher, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Orchestrator{
+		Config:        config,
+		TimeoutConfig: timeoutConfig,
+
+		Queue:          queue,
+		MinibatchStore: minibatchStore,
+		Dispatcher:     dispatcher,
+		EncoderClient:  encoderClient,
+
+		ChainState:            chainState,
+		AssignmentCoordinator: assignmentCoordinator,
+		Aggregator:            aggregator,
+		EncodingStreamer:      encodingStreamer,
+		Transactor:            transactor,
+		TransactionManager:    txnManager,
+		Metrics:               metrics,
+
+		ethClient:      ethClient,
+		finalizer:      finalizer,
+		logger:         logger.With("component", "Orchestrator"),
+		HeartbeatChan:  heartbeatChan,
+		MiniBatcher:    miniBatcher,
+		BatchConfirmer: batchConfirmer,
+	}, nil
+}
+
+func (o *Orchestrator) Start(ctx context.Context) error {
+	err := o.ChainState.Start(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Wait for few seconds for indexer to index blockchain
+	// This won't be needed when we switch to using Graph node
+	time.Sleep(indexerWarmupDelay)
+	err = o.EncodingStreamer.Start(ctx)
+	if err != nil {
+		return err
+	}
+	batchTrigger := o.EncodingStreamer.EncodedSizeNotifier
+
+	err = o.BatchConfirmer.Start(ctx)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		receiptChan := o.TransactionManager.ReceiptChan()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case receiptOrErr := <-receiptChan:
+				o.logger.Info("received response from transaction manager", "receipt", receiptOrErr.Receipt, "err", receiptOrErr.Err)
+				err := o.BatchConfirmer.ProcessConfirmedBatch(ctx, receiptOrErr)
+				if err != nil {
+					o.logger.Error("failed to process confirmed batch", "err", err)
+				}
+			}
+		}
+	}()
+	o.TransactionManager.Start(ctx)
+
+	o.finalizer.Start(ctx)
+
+	go func() {
```

Comment:
Should we keep this loop in minibatcher (and `Start` method)? 
Looks like we moved it here to have `signalLiveness` interweaved in the loop, but signaling liveness from the high level component seems sufficient

Reply:
Don't we need to signalLiveness on some sort of reoccurring loop where work is being done?